checkov/terraform/checks/module/base_module_check.py

```diff
@@ -48,4 +48,6 @@ def scan_module_conf(self, conf: Dict[str, List[Any]]) -> CheckResult:
 
     @staticmethod
     def is_git_source(source: str) -> bool:
-        return source.startswith('git@') or source.startswith('git::') or source.startswith('github.com') or source.startswith('bitbucket.org')
+        parsed_url = urlparse(source)
+        hostname = parsed_url.hostname
+        return hostname in {"github.com", "bitbucket.org"} or source.startswith('git@') or source.startswith('git::')
```

Comment:
no sanitization needed here. the module source value is just checked for a prefix and not used otherwise.

and the code doesn't work as you suggested, because `urlparse(github.com/hashicorp/example).hostname is None`